From 6327533b1fd29fa86f6bf34e61c332c010e3c689 Mon Sep 17 00:00:00 2001
From: Nick Clifton <nickc@redhat.com>
Date: Tue, 17 Apr 2018 14:30:07 +0100
Subject: [PATCH] Add a check for a NULL table pointer before attempting to
 compute a DWARF filename.

	PR 23065
	* dwarf2.c (concat_filename): Check for a NULL table pointer.
---
 bfd/dwarf2.c  | 2 +-
 1 files changed, 1 insertions(+), 1 deletion(-)

diff --git a/bfd/dwarf2.c b/bfd/dwarf2.c
index ca22db7..0f8257f 100644
--- a/bfd/dwarf2.c
+++ b/bfd/dwarf2.c
@@ -1566,7 +1566,7 @@ concat_filename (struct line_info_table *table, unsigned int file)
 {
   char *filename;
 
-  if (file - 1 >= table->num_files)
+  if (table == NULL || file - 1 >= table->num_files)
     {
       /* FILE == 0 means unknown.  */
       if (file)
-- 
2.9.3


